includes/Core/Authentication/Authentication.php

```diff
@@ -1235,6 +1235,17 @@ function clearSiteKitAppStorage() {
 					if ( ! empty( $this->user_options->get( OAuth_Client::OPTION_ERROR_CODE ) ) ) {
 						return false;
 					}
+
+					$unsatisfied_scopes = $this->get_oauth_client()->get_unsatisfied_scopes();
+
+					if (
+						Feature_Flags::enabled( 'gteSupport' )
+						&& count( $unsatisfied_scopes ) === 1
+						&& in_array( 'https://www.googleapis.com/auth/tagmanager.readonly', $unsatisfied_scopes, true )
```

Comment:
A minor point, but why not just test using `===` here? Seems marginally clearer...

```suggestion
						&& 'https://www.googleapis.com/auth/tagmanager.readonly' === $unsatisfied_scopes[0]
```